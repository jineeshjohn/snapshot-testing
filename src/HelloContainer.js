import React from "react";
import styled from 'styled-components';
import Hello from "./Hello";

const Para = styled.div`
  color: red;
  background-color: ${({hasTooltip}) => hasTooltip ? 'blue' :  'green'}
`;


const HelloContainer = props => {
  return (
    <div>
      <Hello {...props} />
      <Para id="123" alt="this is tooltip" hasTooltip={true}>This is test</Para>
    </div>
  );
};

export default HelloContainer;
