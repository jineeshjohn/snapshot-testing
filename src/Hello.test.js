import React from "react";

import Enzyme, { shallow } from "enzyme";
import renderer from 'react-test-renderer';
import Adapter from "enzyme-adapter-react-16";

import HelloContainer from "./HelloContainer";
import Hello from "./Hello";

Enzyme.configure({ adapter: new Adapter() });

it("renders properly", () => {
  const wrapper = renderer.create(<HelloContainer name="CodeSandbox" />);
  expect(wrapper).toMatchSnapshot();
});
