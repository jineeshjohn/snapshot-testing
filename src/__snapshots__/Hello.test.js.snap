// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div>
  <h1>
    Hello 
    CodeSandbox
    !
  </h1>
  <div
    alt="this is tooltip"
    className="sc-bdVaJa bEXVQr"
    id="123"
  >
    This is test
  </div>
</div>
`;
